File: Utilities/DatabaseUpdater/FolderStructureScriptSource.vb
Imports System.Collections.Generic
Public Class FolderStructureScriptSource
	Inherits ScriptSource


	Dim path As String

	Sub New(ByVal path As String)
		Me.path = path
	End Sub

	Public Overrides Sub FillTables()
		Dim dir As New IO.DirectoryInfo(path)
		AddScriptsFromFolder(dir.FullName & "\Change Scripts", "*.sql", True, ScriptType.UpdateScript)
		AddScriptsFromFolder(dir.FullName & "\Data Scripts", "*.sql", True, ScriptType.TableData)
		AddScriptsFromFolder(dir.FullName & "\Create scripts\viw", "*.sql", True, ScriptType.View)
		AddScriptsFromFolder(dir.FullName & "\Create scripts\udf", "*.sql", True, ScriptType.UserDefinedFunction)
		AddScriptsFromFolder(dir.FullName & "\Create scripts\prc", "*.sql", True, ScriptType.StoredProcedure)
		AddScriptsFromFolder(dir.FullName & "\Create scripts\trg", "*.sql", True, ScriptType.Trigger)

	End Sub
	Private Sub AddScriptsFromFolder(ByVal path As String, ByVal extensionMask As String, ByVal recurse As Boolean, ByVal type As String)
		Dim dir As New IO.DirectoryInfo(path)
		If Not dir.Exists Then Exit Sub
		For Each file As IO.FileInfo In dir.GetFiles(extensionMask)
			AddScript(type, file.FullName)
		Next
		If recurse Then
			For Each directory As IO.DirectoryInfo In dir.GetDirectories
				AddScriptsFromFolder(directory.FullName, extensionMask, recurse, type)
			Next
		End If

	End Sub

End Class

File: Utilities/SqlScriptRunner/ScriptSources/FolderStructureScriptSource.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
namespace SqlScriptRunner.ScriptSources
{
    using System.Collections.Generic;
    internal class FolderStructureScriptSource : ScriptSource
    {
        DirectoryInfo dir;

        internal FolderStructureScriptSource(string path)
        {
            
            dir = new DirectoryInfo(path);
			AddScriptsFromFolder(Path.Combine(dir.FullName, ScriptSource.OnSyncScriptDir), "*.sql", true, ScriptType.OnSync);
            AddScriptsFromFolder(Path.Combine(dir.FullName, ScriptSource.UpdateScriptDir), "*.sql", true, ScriptType.UpdateScript);
            AddScriptsFromFolder(Path.Combine(dir.FullName, ScriptSource.DataScriptDir) , "*.sql", true, ScriptType.TableData);
            string createScriptDir = Path.Combine(dir.FullName, ScriptSource.CreateScriptDir);
            AddScriptsFromFolder(Path.Combine(createScriptDir, ScriptSource.ViewSubDir ), "*.sql", true, ScriptType.View);
            AddScriptsFromFolder(Path.Combine(createScriptDir, ScriptSource.FunctionSubDir) , "*.sql", true, ScriptType.UserDefinedFunction);
            AddScriptsFromFolder(Path.Combine(createScriptDir, ScriptSource.ProcedureSubDir) , "*.sql", true, ScriptType.StoredProcedure);
            AddScriptsFromFolder(Path.Combine(createScriptDir, ScriptSource.TriggerSubDir), "*.sql", true, ScriptType.Trigger);
        }

        private void AddScriptsFromFolder(string path, string extensionMask, bool recurse, string type)
        {
            DirectoryInfo dir = new DirectoryInfo(path);
            if (!(dir.Exists))
            {
                return;
            }
            foreach (FileInfo file in dir.GetFiles(extensionMask))
            {
                AddScript(type, file.FullName);
            }
            if (recurse)
            {
                foreach (DirectoryInfo directory in dir.GetDirectories())
                {
                    AddScriptsFromFolder(directory.FullName, extensionMask, recurse, type);
                }
            }
        }
    }
}
